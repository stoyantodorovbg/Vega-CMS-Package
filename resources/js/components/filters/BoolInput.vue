<template>
    <div>
        <label>{{ name }}</label>
        <input @change="changeInput(name, 'exact')"
               v-model="inputValue"
               type="checkbox">
    </div>
</template>

<script>
    export default {
        name: 'BoolInput',

        props: {
            name: {
                type: String,
                default: 'status',
            },
        },

        data() {
            return {
                inputValue: true,
            }
        },

        methods: {
            changeInput(fieldName, type)
            {
                this.$parent.$parent.$parent.updateFilters(fieldName, this.inputValue, type);
            }
        }
    }
</script>
