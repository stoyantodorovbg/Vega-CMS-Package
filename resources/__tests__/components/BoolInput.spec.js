import { mount } from '@vue/test-utils'
import BoolInput from '../../js/components/filters/BoolInput.vue'

describe('BoolInput Component', () => {
  test('should mount without crashing', () => {
    const wrapper = mount(BoolInput);

    expect(wrapper).toMatchSnapshot()
  });
});

