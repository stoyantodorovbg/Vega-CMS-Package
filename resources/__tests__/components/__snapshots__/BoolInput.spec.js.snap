// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoolInput Component should mount without crashing 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
